libvirt/resource_libvirt_domain_test.go

```diff
@@ -490,6 +522,58 @@ func testAccCheckLibvirtDomainDestroy(s *terraform.State) error {
 	return nil
 }
 
+func testAccCheckLibvirtDomainIsDestroyed(n string, domain *libvirt.Domain) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No libvirt domain ID is set")
+		}
+
+		virConn := testAccProvider.Meta().(*Client).libvirt
+
+		retrieveDomain, err := virConn.LookupDomainByUUIDString(rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		log.Printf("The ID is %s", rs.Primary.ID)
+
+		realID, err := retrieveDomain.GetUUIDString()
+		if err != nil {
+			return err
+		}
+
+		if realID != rs.Primary.ID {
+			return fmt.Errorf("Libvirt domain not found")
+		}
+
+		// 1) test that domain is not up
+		domainRuns, err := IsDomainRunning(*domain)
+		if err != nil {
+			return err
+		}
+		if domainRuns {
+			return fmt.Errorf("Domain should be destroyed! TEST FAILED")
+		}
+		// 2 check that status is shutdown
+		state, err := GetDomainState(*domain)
+		if err != nil {
+			return fmt.Errorf("could not get domain state: %s", err)
+		}
+
+		if state != "shutoff" {
+			return fmt.Errorf("Domain state should be == shutoff, but is %s, TEST FAILED", state)
```

Comment:
`Domain state should be 'shutoff' but is '%s'`. Again, no need for `TEST FAILED`.